modules/cnvkit/1.0/cnvkit.smk

```diff
@@ -0,0 +1,318 @@
+    resources: 
+        **CFG["resources"]["SNPs"]
+    group: "cnvkit"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    shell: 
+        """
+            bcftools concat {input.vcf} -Oz -o {output.vcf} && 
+            tabix -p vcf {output.vcf}
+        """
+
+
+# ----------------------------------------------------------------------------------------------- #
+# Integrating cnvkit with BAF
+# ----------------------------------------------------------------------------------------------- #
+rule _run_cnvkit_call_vcf:
+    input: 
+        cns = CFG["dirs"]["cn_calls"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.call.cns",
+        vcf = CFG["dirs"]["SNPs"]  + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.vcf.gz",
+        tbi = CFG["dirs"]["SNPs"]  + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.vcf.gz.tbi"
+    output: 
+        cns =  CFG["dirs"]["BAF"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.call.cns"
+    params:
+        rescale = CFG["options"]["BAF"]["rescale"],
+        min_depth = CFG["options"]["BAF"]["min_depth"]
+    log: 
+        CFG["logs"]["BAF"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}_call.log"
+    conda: 
+        CFG["conda_envs"]["cnvkit"]
+    shell:
+        """
+            cnvkit.py call {input.cns} --output {output.cns} -v {input.vcf} --min-variant-depth {params.min_depth} -m {params.rescale} &> {log}
+        """
+
+
+rule _run_cnvkit_scatter:
+    input:
+        cnr = CFG["dirs"]["cnvkit"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.cnr",
+        cns = CFG["dirs"]["BAF"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.call.cns",
+        vcf = CFG["dirs"]["SNPs"]  + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.vcf.gz",
+        tbi = CFG["dirs"]["SNPs"]  + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.vcf.gz.tbi"
+    output: 
+        png = CFG["dirs"]["scatter"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}_scatter.png"
+    params:
+        min_depth = CFG["options"]["scatter"]["min_depth"],
+        ymax = CFG["options"]["scatter"]["ymax"],
+        ymin = CFG["options"]["scatter"]["ymin"]
+    conda: 
+        CFG["conda_envs"]["cnvkit"]
+    shell:
+        "cnvkit.py scatter {input.cnr} -s {input.cns} --output {output.png} -v {input.vcf} --min-variant-depth {params.min_depth} --y-max {params.ymax} --y-min {params.ymin}"
+
+
+rule _cnvkit_to_seg:
+    input:
+        expand(
+            [
+                CFG["dirs"]["BAF"] + "{{seq_type}}--{{genome_build}}{{masked}}/{capture_space}/{tumour_id}.call.cns"
+            ],
+            zip,  # Run expand() with zip(), not product()
+            tumour_id=CFG["runs"]["tumour_sample_id"],
+            capture_space=CFG["runs"]["tumour_capture_space"])
+    output:
+        CFG["dirs"]["outputs"] + "seg/{seq_type}--{genome_build}{masked}/cnvkit.seg"
```

Comment:
yes cnvkit has an option to just neatly compile all the samples into one seg

Reply:
Is it possible to have individual seg-files? Imagine adding these to MySQL database, or having some downstream tool that needs a path to a per-sample file, this may be hard to deal with